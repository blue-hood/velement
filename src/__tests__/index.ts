import { getByText } from '@testing-library/dom';
import '@testing-library/jest-dom/extend-expect';

import VirtualElement, { createElement } from '..';

test('should create HTMLElement', async () => {
    const div = createElement<null, HTMLDivElement>('div', null);
    div.innerHTML = `HTMLDivElement`;

    expect(div).toMatchSnapshot();
});

test('should create VirtualElement', async () => {
    class Div extends VirtualElement<HTMLDivElement> {
        public constructor(element: HTMLDivElement | null) {
            super(element || 'div');

            this.element.innerHTML = 'VirtualDivElement';
        }
    }

    const div = createElement<null, Div>(Div, null);

    expect(div.element).toMatchSnapshot();
});
