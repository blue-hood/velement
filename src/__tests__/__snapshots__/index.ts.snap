// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create HTMLElement 1`] = `
<div>
  HTMLDivElement
</div>
`;

exports[`should create VirtualElement 1`] = `
<div>
  VirtualDivElement
</div>
`;
